File: server/datasets/python_source-code/Multiple-A2/organic6_bubble_sort.py
"""
Bubble Sort - organic6
Approach: Entirely index-driven using nested while loops instead of
for loops.
"""


def bubble_sort(arr):
    n = len(arr)
    i = 0
    while i < n - 1:
        j = 0
        while j < n - 1 - i:
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
            j += 1
        i += 1
    return arr


if __name__ == "__main__":
    ARR = [64, 34, 25, 12, 22, 11, 90, 5, 77, 1]
    print("Sorted:", bubble_sort(ARR))

File: server/datasets/java_source-code/Multiple-A2/organic6_bubble_sort.java
/*
 * Bubble Sort - organic6
 * Approach: Entirely index-driven using nested while loops instead of for loops.
 */

import java.util.Arrays;

class BubbleSort {
    public static int[] bubbleSort(int[] arr) {
        int n = arr.length;
        int i = 0;
        while (i < n - 1) {
            int j = 0;
            while (j < n - 1 - i) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
                j++;
            }
            i++;
        }
        return arr;
    }

    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 90, 5, 77, 1};
        System.out.println("Sorted: " + Arrays.toString(bubbleSort(arr)));
    }
}
